pool/context_pool.go

```diff
@@ -23,6 +23,18 @@ type ContextPool struct {
 // are busy, a call to Go() will block until the task can be started.
 func (p *ContextPool) Go(f func(ctx context.Context) error) {
 	p.errorPool.Go(func() error {
+		if p.cancelOnError {
+			// If we are cancelling on error, then we also want to cancel if a
+			// panic is raised. To do this, we need to recover, cancel, and then
+			// re-throw the caught panic.
+			defer func() {
+				if r := recover(); r != nil {
+					p.cancel()
+					panic(r)
+				}
+			}()
+		}
```

Comment:
Should we add this to `ResultContextPool` too?

Reply:
`ResultContextPool` just uses `ContextPool` internally, so the handling should hold there - I added a test case in e8fb0351c1e3de8c9ad124bfdb2b6362bf0b6d90

Reply:
Oh, duh. I forgot that this part was reused. Thanks for the test!